File: Primes/Primes.hs
----
-- Primes.hs
-- by Daniel Brice
-- Invoke as `Primes <num>`
-- Prints primes not exceeding <num>
----
import System.Environment (getArgs)

isPrime :: Integer -> Bool
isPrime n = case n of
  1 -> False
  2 -> True
  n -> null . dropWhile (0 /=) $ remainders
    where
      remainders = map (rem n) . enumFromTo 2
                 . ceiling . sqrt . fromInteger $ n

primes :: [Integer]
primes = filter isPrime $ [1..]

main :: IO ()
main = do
  args <- getArgs
  let n = read . head $ args
  mapM_ print $ takeWhile (<= n) primes

File: Primes/Primes.py
###
# Primes.py
# by Daniel Brice
# Invoke as `python3 Primes.py <num>`
# Prints primes not exceeding <num>
###
import sys
import math

def isPrime(n):
  if n == 1:
    return False
  if n == 2:
    return True

  ceil = math.ceil(math.sqrt(n))
  for i in range(2, ceil + 1):
    if n % i == 0:
      return False

  return True

n = int(sys.argv[1])
primes = list(filter(isPrime, range(1, n + 1)))
out = "\n".join(list(map(str,primes)))
print(out)
